### ISSUE TITLE
[Bug]: Fix a variety of color contrast accessibility issues for 10.1

### ISSUE THREAD

**Issue by MichaelArestad** [storybookjs/storybook#32961]
### Describe the bug

Most of these are quick and this is just for tracking

* [ ] **Checkbox:** border is below 3.0 with external and internal BG 
* [ ] **Radio:** border is below 3.0 with external and internal BG
* [ ] **Range:** border is below 3.0 with external and internal BG
* [ ] **Interaction panel:** "not" keyword in code highlights is below 4.5
* [ ] **SelectOption** focused + selected state has less than 4.5 between BG and text
* [ ] **A11y violations:** blue "button" for the selected element has too low BG/text contrast
* [ ] **Status colors:** Lighten status colors in dark theme

### Reproduction link

Nah

### System

```bash
Nunya
```

### Additional context

CC @Sidnioulz
